### PR TITLE
Recursive logit model

### PR DESCRIPTION
**Pull request by pabloguarda**

### CODE REVIEW

File: examples/PG_console_testing.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
+# =============================================================================
+# Imports
+# =============================================================================
+
+# Internal modules
+from src.inverse_astar_search.network import *
```
Comment (mbattifarano) [inline examples/PG_console_testing.py]: `from inverse_astar_search.network [...]`

File: requirements/dev.txt [hunk 2]
```diff
@@ -1 +1,4 @@
-pytest==5.4.1
+# pip install -r requirements/dev.txt
```
Comment (mbattifarano) [inline requirements/dev.txt]: remove

File: setup.py [hunk 3]
```diff
@@ -6,7 +6,7 @@ def read_requirements(filename):
         return fp.read().strip().splitlines()
 
 setup(
-    name="inverse-astar-search",
+    name="inverse-pastar.py-search",
```
Comment (mbattifarano) [inline setup.py]: revert

File: src/inverse_astar_search/pastar.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    #     np.linalg.norm(np.array(pos_nodes[edge[0]]) - np.array(pos_nodes[edge[1]]))
+    #
+    # pos_nodes = nx.get_node_attributes(G, nodes_coordinate_label)
+    # len_edges = {}
+    # for edge in G.edges():
+    #     len_edges[edge] = np.linalg.norm(np.array(pos_nodes[edge[0]]) - np.array(pos_nodes[edge[1]]))
+    #
+    # nx.set_edge_attributes(G, len_edges, attribute_label)
+
+    return G
```
Comment (mbattifarano) [inline src/inverse_astar_search/pastar.py]: `raise NotImplemented`

File: src/inverse_astar_search/pastar.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    # g0 = network_lecture_AI(index=0)
+    #
+    # pos = {'s': (0, 0), 'a': (5, 5), 'b': (5, -5), 't': (10, 0)}
+    #
+    # for n, p in pos.items():
+    #     g0.nodes[n]['pos'] = p
+
+    # show_network(g0)
+
+    pass
```
Comment (mbattifarano) [inline src/inverse_astar_search/pastar.py]: remove

File: test/test_LP1.py [hunk 6]
```diff
@@ -0,0 +1,119 @@
+# from inverse_astar_search.optim import linear_program, get_by_node_pair, entropy
```
Comment (mbattifarano) [inline test/test_LP1.py]: remove file

File: test/test_logit.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import hypothesis
```
Comment (mbattifarano) [inline test/test_logit.py:1]: remove file

File: test/test_pastar.py [hunk 8]
```diff
@@ -0,0 +1,88 @@
+import networkx as nx
+
+from inverse_astar_search.pastar import max_admissible_heuristic, heuristic_bounds
+
+
+def test_network_lecture_AI():
```
Comment (mbattifarano) [inline test/test_pastar.py]: put this in `conftest.py` as `network_lecture_AI`

File: test/test_pastar.py [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    g2.add_edge('a', 'c', weight=1)
+    g2.add_edge('b', 'c', weight=2)
+    g2.add_edge('c', 't', weight=3)
+
+    networks.append(g2)
+
+    return networks
+
+
+def test_astar_admissibility():
```
Comment (mbattifarano) [inline test/test_pastar.py]: `def test_astar_admissibility(network_lecture_AI)`

File: test/test_pastar.py [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+    def astar_heuristic(a, b):
+        return max_admissible_heuristic(G=g1, target=b)[a]
+
+    astar_optimal_path = nx.astar_path(G=g1, source='s', target='t', heuristic=astar_heuristic)
+
+    assert optimal_path == astar_optimal_path
+
+
+def test_heuristic_bounds():
```
Comment (mbattifarano) [inline test/test_pastar.py]: remove

File: test/test_pastar.py (hunk 8 above)
Comment (mbattifarano) [reply test/test_pastar.py]: `params=[0, 1]` in call to `fixture`
```
@pytest.fixture(params=[0, 1])
def network_lecture_AI(request):
    [...]
    return networks[request.param]
```

**Review by mbattifarano — CHANGES_REQUESTED** [review]

File: test/test_logit.py (hunk 7 above)
Comment (pabloguarda) [reply test/test_logit.py:1]: Tests need to be implemented for the methods included in logit.py. I just removed the content of the script but keep the file for further reference.

**Comment by pabloguarda** [comment]
I am done and ready to merge our pulls to master.

**Review by mbattifarano — APPROVED** [review]

**Pull request merged**